classes/article/ArticleHTMLGalley.inc.php

```diff
@@ -33,6 +34,51 @@ function isHTMLGalley() {
 	}
 
 	/**
+	* Eirik Hanssen, Oslo and Akershus University College of Applied Sciences
+	* Return string contains the <body> element of the HTML Galley file renamed to a <div> element.
+	* The returned string is a html fragment where empty tags will be self closed, as required by xhtml rulesybe.
+	* The purpose is to allow to allow OJS pages displaying the HTML Galley of an article to validate. Without filtering the HTML file string
+	* through the getHTMLBodyContents() function, a valid (full) HTML document will simply be inserted in a <div> element in the OJS page,
+	* and then that resulting webpage won't be valid HTML since it is illegal to have a full html-document inside a div tag.
+	* This function requires the php DOMDocument module to operate. If DOMDocument module is not available, (often part of a php-xml package),
+	* a warning in form of a html-comment is appended to the html string instead of performing this filtering.
+	* @return string
+	*/
+	function getHTMLBodyContents($htmldoc){
+		// first check if DOMDocument class is available
+		if (class_exists('DOMDocument')) {
+			$originalHtmlDocument = new DOMDocument;
+			$htmlBodyContents = new DOMDocument;
+			$originalHtmlDocument->loadHTML($htmldoc);
```

Comment:
After feeding it with several potentially problematic test html galleys, It seems that domDocument->loadHtml() is quite robust. I haven't been able to make it fail parsing, even if the structure is totally messed up. Of course, if the structure is all messed up, that will show in the output. However there might be some gotchas with character sets. I need to investigate a little more regarding to different charager sets. http://php.net/manual/en/domdocument.loadhtml.php